src/lib/OpenSkyApi.ts

```diff
@@ -126,6 +151,34 @@ export class OpenSkyApi {
     }
   }
 
+  private async getOpenSkyFlightsByAircraft(
+    url: string,
+    nvps: Array<Record<string, string>>,
+  ) {
+    const params = new URLSearchParams()
+
+    nvps.forEach((i) => {
+      for (const [key, value] of Object.entries(i)) {
+        params.append(key, value)
+      }
+    })
+
+    try{
+      const data = await this._axios.get<Flight[]>(
+        url,
+        {
+          params,
+        },
+      )
+
+      if(Array.isArray(data?.data)){
+        return data.data;
+      }
+    } catch(x) {}
+    
+    return new Array<Flight>();
```

Comment:
I'd avoid the try/catch here and let application code handle errors the way they deem appropriate. 

It would be more consistent with the rest of the code.

Reply:
Normally, I would agree, but what I discovered is that this endpoint returns a 404 on a valid, but empty set.  Does knowing that change how you would like this handled. Ideally, I think I would catch a 404 specifically and return an empty array, but let it throw the rest to the client.  Thoughts?

Reply:
Good catch! I propose to use axios `validateStatus` config option to handle 404 as a success instead of throwing an error. https://github.com/axios/axios#handling-errors

Reply:
Sounds good to me.  I'll take a look at this over the weekend.

Reply:
Thanks! Let me know if I can help.

Reply:
Ok, I've removed the try/catch and used ValidateStatus to properly catch and deal with the 404 empty set.  From my view of this pull request, it just says "outdated".  So I don't know if you can see the new code or not.  Let me know if you need me to put in a new PR.